api_tests/src/test_environment.ts

```diff
@@ -0,0 +1,116 @@
+import { Script } from "vm";
+import { Config } from "@jest/types";
+import { execSync } from "child_process";
+import { LedgerRunner } from "../lib/ledgers/ledger_runner";
+import { HarnessGlobal } from "../lib/utils";
+import NodeEnvironment from "jest-environment-node";
+import rimraf from "rimraf";
+import * as fs from "fs";
+import { Mutex } from "async-mutex";
+
+// ************************ //
+// Setting global variables //
+// ************************ //
+
+interface ConfigInterface {
+    ledgers: string[];
+    logDir: string;
+}
+
+export class E2ETestEnvironment extends NodeEnvironment {
+    private docblockPragmas: string;
+    private projectRoot: string;
+    private testRoot: string;
+    private logDir: string;
+    private ledgerRunner: LedgerRunner;
+    // @ts-ignore
+    public global: HarnessGlobal;
+
+    constructor(config: Config.ProjectConfig, context: any) {
+        super(config);
+
+        this.docblockPragmas = context.docblockPragmas;
+
+        this.projectRoot = execSync("git rev-parse --show-toplevel", {
+            encoding: "utf8",
+        }).trim();
+        this.testRoot = this.projectRoot + "/api_tests";
+    }
+
+    async setup() {
+        await super.setup();
+        // setup global variables
+        this.global.projectRoot = this.projectRoot;
+        this.global.testRoot = this.testRoot;
+        this.global.ledgerConfigs = {};
+        this.global.verbose =
+            this.global.process.argv.find(item => item.includes("verbose")) !==
+            undefined;
+        this.logDir = "unspecified";
+        this.global.parityAccountMutex = new Mutex();
+
+        if (this.global.verbose) {
+            console.log(`Starting up test environment`);
+        }
+
+        // Will trigger if docblock contains @ledgers
+        // @ts-ignore
+        const configString = this.docblockPragmas.config;
+
+        if (configString) {
+            const config: ConfigInterface = JSON.parse(configString);
+            const logDir = config.logDir ? config.logDir : "unspecified";
```

Comment:
I am guessing it's old code? I think we should assume/check that `config.logDir` is indeed specified and let the framework fail down the line if not.

Reply:
I'll provide an updated version in the next push.